### ISSUE TITLE
"cannot handle constraint: bool_clause" using var if condition with MIP backend at O2

### ISSUE THREAD

**Issue by sbronson** [MiniZinc/libminizinc#629]
```
var bool: b;
var 0..1: x = if b then 1 else 0 endif;
```

```
$ minizinc --solver org.minizinc.mip.coin-bc -O2 f.mzn
MiniZinc has encountered an internal error. This is a bug.
Please file a bug report using the MiniZinc bug tracker.
The internal error message was:
"Error: solver backend cannot handle constraint: bool_clause"
```

On develop.

**Comment by a1880** [comment]
In MiniZinc IDE 2.6.4 under Windows 10, I cannot reproduce this error.
I have tried various backends including CPLEX, SCIP, OR Tools, and COIN-BC 2.10.8/1.17.7 (without MIP!) with -O1 and -O2.

The nearest are Couenne 0.5.7 or Bonmin 1.8.7 as backend:

```
Message:
Builtins bool_clause not implemented or not recognized.

```
The FlatZinc code generated for Couenne:

```
var bool: b:: output_var;
var 0..1: x:: is_defined_var;
var bool: X_INTRODUCED_6_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_7_ ::var_is_introduced :: is_defined_var;
constraint bool_clause([X_INTRODUCED_6_],[b]);
constraint array_bool_or([b,X_INTRODUCED_7_],true);
constraint int_eq_reif(x,1,X_INTRODUCED_6_):: defines_var(X_INTRODUCED_6_);
constraint int_eq_reif(x,0,X_INTRODUCED_7_):: defines_var(X_INTRODUCED_7_);
solve  satisfy;

```

**Comment by Dekker1** [comment]
This likely is a problem with the multi-pass optimisations. Note that the problem only occurs when this is activated. @lokdlok Could you have a look at this?

@a1880 This is different from the NL/Couenne interface, which currently doesn't implement all the necessary decompositions. Even without multi-pass optimisations, this might throw errors when some parts of MiniZinc are used.

**Comment by cyderize** [comment]
This should now be fixed in version 2.7.0
